### PR TITLE
Union NULL types cast to same type as neighbor

### PR DESCRIPTION
**Pull request by max-hoffman**
3+-way unions are subject to a bug where a NULL projection will be cast to a char type, and then have a type clash with the subsequent table relation. Rearranging the query to have the NULL projection last appears to fix the bug because the `convert` cast  is moved to a position where it cannot have a type clash with a table relation to the right.

### CODE REVIEW

File: enginetest/queries/queries.go [hunk 1]
```diff
@@ -742,6 +742,23 @@ var SpatialQueryTests = []QueryTest{
 }
 
 var QueryTests = []QueryTest{
+	{
+		Query: "select * from (select 1, 1) a(x,y) union select * from (select 1, NULL) b(x,y) union select * from (select 1,1) c(x,y);",
```
Comment (zachmu) [inline enginetest/queries/queries.go:746]: Seems like the patch below is only going to work with literal NULL values (nothing else is types.Null)

Need a test where you're doing the union selects from table values that have null in them

**Review by zachmu — COMMENTED** [review]
Need moar testing

File: enginetest/queries/queries.go (hunk 1 above)
Comment (max-hoffman) [reply enginetest/queries/queries.go:746]: i think that's unrelated? a nullable column won't have sql.NullType

**Review by zachmu — APPROVED** [review]
LGTM
